app/assets/javascripts/services/alerts_center_service.js

```diff
@@ -0,0 +1,828 @@
+    var deferred = $q.defer();
+    var resourceOptions = '?expand=resources,alert_actions&attributes=assignee,resource';
+    var limitOptions = '';
+    var offsetOptions = '';
+    var sortOptions = '';
+
+    if (sortField) {
+      sortOptions = '&sort_by=' + sortField + '&sort_order=' + (sortAscending ? 'asc' : 'desc');
+    }
+
+    if (limit) {
+      limitOptions = '&limit=' + limit;
+    }
+
+    if (offset) {
+      offsetOptions = '&offset=' + offset;
+    }
+
+    // Get the alert data
+    API.get(alertsURL + resourceOptions + limitOptions + offsetOptions + sortOptions).then(function(response) {
+      deferred.resolve(response);
+    });
+
+    return deferred.promise;
+  };
+
+  function getObjectType(item) {
+    var objectType = item.type;
+    var descriptors = item.type.split("::");
+
+    if (descriptors.length >= 3) {
+      objectType = descriptors[2];
+    }
+
+    return objectType;
+  }
+
+  function convertApiTime(apiTimestamp) {
+    var apiDate = new Date(apiTimestamp);
+    return apiDate.getTime();
+  }
+
+  function updateAlertStatus(updateAlert) {
+    if (updateAlert && updateAlert.alert_actions && updateAlert.alert_actions.length > 0) {
+      var i;
+      var actionUser;
+
+      for (i = 0; i < updateAlert.alert_actions.length; i++) {
+        updateAlert.alert_actions[i].created_at = convertApiTime(updateAlert.alert_actions[i].created_at);
+        updateAlert.alert_actions[i].updated_at = convertApiTime(updateAlert.alert_actions[i].updated_at);
+      }
+
+      // Sort from newest to oldest
+      updateAlert.alert_actions.sort(function(state1, state2) {
+        return state2.updated_at - state1.updated_at;
+      });
+
+      // Set the lastUpdate to the time of the newest state change
+      updateAlert.lastUpdate = updateAlert.alert_actions[0].updated_at;
+
+      // update each state
+      updateAlert.numComments = 0;
+      for (i = 0; i < updateAlert.alert_actions.length; i++) {
+        actionUser = _this.getUserByIdOrUserId(updateAlert.alert_actions[i].user_id);
+        updateAlert.alert_actions[i].username = angular.isDefined(actionUser) ? actionUser.name : '';
+
+        // Bump the comments count if a comment was made
+        if (updateAlert.alert_actions[i].comment) {
+          updateAlert.numComments++;
+        }
+      }
+
+      if (updateAlert.numComments === 1) {
+        updateAlert.commentsTooltip = sprintf(__("%d Note"), 1);
+      } else {
+        updateAlert.commentsTooltip = sprintf(__("%d Notes"), updateAlert.numComments);
+      }
+    }
+  }
+
+  function convertAlert(alertData, key, objectName, objectType, retrievalTime) {
+    var path = '/assets/svg/';
+    var suffix = '.svg';
+    var prefix = '';
+    var imageName = objectType.replace(/([a-z\d])([A-Z]+)/g, '$1_$2').replace(/[-\s]+/g, '_').toLowerCase();
+
+    if (key === 'providers') {
+      prefix = 'vendor-';
+    } else {
+      prefix = 'os-';
+    }
+
+    var typeImage = path + prefix + imageName + suffix;
```

Comment:
Nope nope nope, this will never work in production, sorry.

The full image path needs to come from the backend, and needs to pass through the `image_path` helper, otherwise, you'll get an invalid URL in production mode.